BrowserKit/Tests/SummarizeKitTests/SummarizeControllerTests.swift

```diff
@@ -92,20 +92,24 @@ final class SummarizeControllerTests: XCTestCase {
 
     override func setUp() {
         super.setUp()
-        UIView.setAnimationsEnabled(false)
-        summarizer = MockSummarizer(shouldRespond: ["Response"], shouldThrowError: nil)
-        navigationHandler = MockSummarizeNavigationHandler()
-        webView = MockWebView(URL(string: "https://www.example.com")!)
-        viewModel = MockSummarizeViewModel()
-        AppContainer.shared.register(service: DefaultThemeManager(sharedContainerIdentifier: "") as ThemeManager)
+        ensureMainThread {
+            UIView.setAnimationsEnabled(false)
+            self.summarizer = MockSummarizer(shouldRespond: ["Response"], shouldThrowError: nil)
+            self.navigationHandler = MockSummarizeNavigationHandler()
+            self.webView = MockWebView(URL(string: "https://www.example.com")!)
+            self.viewModel = MockSummarizeViewModel()
+            AppContainer.shared.register(service: DefaultThemeManager(sharedContainerIdentifier: "") as ThemeManager)
+        }
     }
 
     override func tearDown() {
```

Comment:
I think using  `override func tearDown() async throws {` instead here will save us to use the `ensureMainThread` call